core/renderers/common/path_object.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * @license
+ * Visual Blocks Editor
+ *
+ * Copyright 2016 Google Inc.
+ * https://developers.google.com/blockly/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview TODO
+ * @author fenichel@google.com (Rachel Fenichel)
+ */
+
+'use strict';
+
+goog.provide('Blockly.blockRendering.IPathObject');
+goog.provide('Blockly.blockRendering.PathObject');
+
+goog.require('Blockly.utils.dom');
+
+/**
+ * An interface for a block's path object.
+ * @param {!SVGElement} _root The root SVG element.
+ * @interface
+ * @package
+ */
+Blockly.blockRendering.IPathObject = function(_root) {};
+
+/**
+ * An object that handles creating and setting each of the SVG elements
+ * used by the renderer.
+ * @param {!SVGElement} root The root SVG element.
+ * @constructor
+ * @implements {Blockly.blockRendering.IPathObject}
+ * @package
+ */
+Blockly.blockRendering.PathObject = function(root) {
+  this.svgRoot = root;
+
+  /**
+   * @type {SVGElement}
+   * @package
+   */
+  this.svgPath = Blockly.utils.dom.createSvgElement('path',
+      {'class': 'blocklyPath'}, this.svgRoot);
+
+  // The light and dark paths need to exist (for now) because there is colouring
```

Comment:
Is this still true?

Reply:
Yeah. We're still assume these paths exist in [block_svg::setBorderColor_](https://github.com/google/blockly/blob/develop/core/block_svg.js#L995)